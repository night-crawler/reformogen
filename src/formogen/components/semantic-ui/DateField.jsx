import React from 'react';
import { Form } from 'semantic-ui-react';
import DatePicker from 'react-datepicker';
import moment from 'moment';

import { MessageList } from './MiscComponents';
import _ from 'lodash';

import Label from './Label';

import propTypes from '../fieldPropTypes';


DateField.propTypes = propTypes;
export default function DateField(props) {
    const handleChange = (momentTimeObject, e) => {
        // proxy event with SUI-React compliant style
        props.onChange(e, {
            name: props.name,
            value: momentTimeObject ? momentTimeObject.toISOString() : null
        });
    };

    let _props = {
        name: props.name,
        selected: props.value ? moment(props.value) : null,  // null == no selected value
        onChange: handleChange,
        showTimeSelect: false,
        locale: props.locale,
        placeholderText: props.placeholder,
        todayButton: 'Now',
        dateFormat: 'YYYY.MM.DD',
        showYearDropdown: true,
        showMonthDropdown: true,
        dropdownMode: 'select',
    };

    if (_.isFunction(props.updateProps)) {
        _props = props.updateProps(_props, props);
    }

    return (
        <Form.Field
            required={ props.required }
            disabled={ !props.editable }
            width={ props.layoutOpts.width }
            error={ !_.isEmpty(props.errors) }
        >
            <Label { ...props } />
            <DatePicker { ..._props } />
            { !props.helpTextOnHover ? <span className="help-text">{ props.help_text }</span> : ''}
            <MessageList messages={ props.errors } />
        </Form.Field>
    );
}
