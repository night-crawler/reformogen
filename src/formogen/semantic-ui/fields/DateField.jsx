import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import moment from 'moment';
import { Form } from 'semantic-ui-react';
import DatePicker from 'react-datepicker';

import { errorsType, layoutOptsType } from '../../fieldPropTypes';
import Label from '../common/Label';
import ErrorsList from '../common/ErrorsList';


DateField.propTypes = {
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  max_length: PropTypes.number,
  help_text: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  errors: errorsType,

  required: PropTypes.bool,
  editable: PropTypes.bool,

  helpTextOnHover: PropTypes.bool,
  layoutOpts: layoutOptsType,
  locale: PropTypes.string,

  updateProps: PropTypes.func,
  onChange: PropTypes.func,
};
export default function DateField(props) {
  const handleChange = (momentTimeObject, e) => {
    // proxy event with SUI-React compliant style
    props.onChange(e, {
      name: props.name,
      value: momentTimeObject ? momentTimeObject.toISOString() : null
    });
  };

  let _props = {
    name: props.name,
    selected: props.value ? moment(props.value) : null,  // null == no selected value
    onChange: handleChange,
    showTimeSelect: false,
    locale: props.locale,
    placeholderText: props.placeholder,
    todayButton: 'Now',
    dateFormat: 'YYYY.MM.DD',
    showYearDropdown: true,
    showMonthDropdown: true,
    dropdownMode: 'select',
  };

  if (_.isFunction(props.updateProps)) {
    _props = props.updateProps(_props, props);
  }

  return (
    <Form.Field
      required={ props.required }
      disabled={ !props.editable }
      width={ props.layoutOpts.width }
      error={ !_.isEmpty(props.errors) }
    >
      <Label { ...props } />
      <DatePicker { ..._props } />
      {
        !props.helpTextOnHover
          ? <span className='help-text'>{ props.help_text }</span>
          : ''
      }
      <ErrorsList messages={ props.errors } />
    </Form.Field>
  );
}
