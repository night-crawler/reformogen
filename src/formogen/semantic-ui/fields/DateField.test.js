import { DateField } from './DateField';


it('<DateField />', () => {
  const wrapper = shallow(
    <DateField
      type='DateField'
      name='DateField: name'
      verbose_name='DateField: verbose_name'
      help_text='DateField: help_text'
      displayOptions={ { width: 1 } }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
