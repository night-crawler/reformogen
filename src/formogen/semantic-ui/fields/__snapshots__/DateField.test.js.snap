// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateField /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DateField
    dateFormat="YYYY.MM.DD"
    displayOptions={
      Object {
        "width": 1,
      }
    }
    dropdownMode="select"
    help_text="DateField: help_text"
    name="DateField: name"
    showMonthDropdown={true}
    showTimeSelect={false}
    showYearDropdown={true}
    todayButton="Now"
    type="DateField"
    verbose_name="DateField: verbose_name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <FieldLabel
          dateFormat="YYYY.MM.DD"
          displayOptions={
            Object {
              "width": 1,
            }
          }
          dropdownMode="select"
          help_text="DateField: help_text"
          name="DateField: name"
          showMonthDropdown={true}
          showTimeSelect={false}
          showYearDropdown={true}
          todayButton="Now"
          type="DateField"
          verbose_name="DateField: verbose_name"
        />,
        <DatePicker
          allowSameDay={false}
          dateFormat="YYYY.MM.DD"
          dateFormatCalendar="MMMM YYYY"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="select"
          monthsShown={1}
          name="DateField: name"
          nextMonthButtonLabel="Next month"
          onBlur={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          readOnly={false}
          selected={null}
          shouldCloseOnSelect={true}
          showMonthDropdown={true}
          showTimeSelect={false}
          showYearDropdown={true}
          style={
            Object {
              "width": "100%",
            }
          }
          timeCaption="Time"
          timeIntervals={30}
          todayButton="Now"
          withPortal={false}
        />,
        <span
          className="help-text"
        >
          DateField: help_text
        </span>,
        <ErrorsList
          color="red"
          header="Errors"
        />,
      ],
      "disabled": true,
      "error": false,
      "required": undefined,
      "width": 1,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "dateFormat": "YYYY.MM.DD",
          "displayOptions": Object {
            "width": 1,
          },
          "dropdownMode": "select",
          "help_text": "DateField: help_text",
          "name": "DateField: name",
          "showMonthDropdown": true,
          "showTimeSelect": false,
          "showYearDropdown": true,
          "todayButton": "Now",
          "type": "DateField",
          "verbose_name": "DateField: verbose_name",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowSameDay": false,
          "dateFormat": "YYYY.MM.DD",
          "dateFormatCalendar": "MMMM YYYY",
          "disabled": false,
          "disabledKeyboardNavigation": false,
          "dropdownMode": "select",
          "locale": undefined,
          "monthsShown": 1,
          "name": "DateField: name",
          "nextMonthButtonLabel": "Next month",
          "onBlur": [Function],
          "onChange": [Function],
          "onClickOutside": [Function],
          "onFocus": [Function],
          "onInputClick": [Function],
          "onKeyDown": [Function],
          "onMonthChange": [Function],
          "onSelect": [Function],
          "onYearChange": [Function],
          "placeholderText": undefined,
          "preventOpenOnFocus": false,
          "previousMonthButtonLabel": "Previous Month",
          "readOnly": false,
          "selected": null,
          "shouldCloseOnSelect": true,
          "showMonthDropdown": true,
          "showTimeSelect": false,
          "showYearDropdown": true,
          "style": Object {
            "width": "100%",
          },
          "timeCaption": "Time",
          "timeIntervals": 30,
          "todayButton": "Now",
          "withPortal": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "DateField: help_text",
          "className": "help-text",
        },
        "ref": null,
        "rendered": "DateField: help_text",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "red",
          "header": "Errors",
          "messages": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <FieldLabel
            dateFormat="YYYY.MM.DD"
            displayOptions={
              Object {
                "width": 1,
              }
            }
            dropdownMode="select"
            help_text="DateField: help_text"
            name="DateField: name"
            showMonthDropdown={true}
            showTimeSelect={false}
            showYearDropdown={true}
            todayButton="Now"
            type="DateField"
            verbose_name="DateField: verbose_name"
          />,
          <DatePicker
            allowSameDay={false}
            dateFormat="YYYY.MM.DD"
            dateFormatCalendar="MMMM YYYY"
            disabled={false}
            disabledKeyboardNavigation={false}
            dropdownMode="select"
            monthsShown={1}
            name="DateField: name"
            nextMonthButtonLabel="Next month"
            onBlur={[Function]}
            onChange={[Function]}
            onClickOutside={[Function]}
            onFocus={[Function]}
            onInputClick={[Function]}
            onKeyDown={[Function]}
            onMonthChange={[Function]}
            onSelect={[Function]}
            onYearChange={[Function]}
            preventOpenOnFocus={false}
            previousMonthButtonLabel="Previous Month"
            readOnly={false}
            selected={null}
            shouldCloseOnSelect={true}
            showMonthDropdown={true}
            showTimeSelect={false}
            showYearDropdown={true}
            style={
              Object {
                "width": "100%",
              }
            }
            timeCaption="Time"
            timeIntervals={30}
            todayButton="Now"
            withPortal={false}
          />,
          <span
            className="help-text"
          >
            DateField: help_text
          </span>,
          <ErrorsList
            color="red"
            header="Errors"
          />,
        ],
        "disabled": true,
        "error": false,
        "required": undefined,
        "width": 1,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "dateFormat": "YYYY.MM.DD",
            "displayOptions": Object {
              "width": 1,
            },
            "dropdownMode": "select",
            "help_text": "DateField: help_text",
            "name": "DateField: name",
            "showMonthDropdown": true,
            "showTimeSelect": false,
            "showYearDropdown": true,
            "todayButton": "Now",
            "type": "DateField",
            "verbose_name": "DateField: verbose_name",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowSameDay": false,
            "dateFormat": "YYYY.MM.DD",
            "dateFormatCalendar": "MMMM YYYY",
            "disabled": false,
            "disabledKeyboardNavigation": false,
            "dropdownMode": "select",
            "locale": undefined,
            "monthsShown": 1,
            "name": "DateField: name",
            "nextMonthButtonLabel": "Next month",
            "onBlur": [Function],
            "onChange": [Function],
            "onClickOutside": [Function],
            "onFocus": [Function],
            "onInputClick": [Function],
            "onKeyDown": [Function],
            "onMonthChange": [Function],
            "onSelect": [Function],
            "onYearChange": [Function],
            "placeholderText": undefined,
            "preventOpenOnFocus": false,
            "previousMonthButtonLabel": "Previous Month",
            "readOnly": false,
            "selected": null,
            "shouldCloseOnSelect": true,
            "showMonthDropdown": true,
            "showTimeSelect": false,
            "showYearDropdown": true,
            "style": Object {
              "width": "100%",
            },
            "timeCaption": "Time",
            "timeIntervals": 30,
            "todayButton": "Now",
            "withPortal": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "DateField: help_text",
            "className": "help-text",
          },
          "ref": null,
          "rendered": "DateField: help_text",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "red",
            "header": "Errors",
            "messages": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
